features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64LRUCache.kt

```diff
@@ -0,0 +1,134 @@
+    @Suppress("MagicNumber")
+    val MAX_CACHE_MEMORY_SIZE_BYTES = 4 * 1024 * 1024 // 4MB
+    @Suppress("MagicNumber")
+    private val ON_LOW_MEMORY_SIZE_BYTES = MAX_CACHE_MEMORY_SIZE_BYTES / 2 // 50% size
+    @Suppress("MagicNumber")
+    private val ON_MODERATE_MEMORY_SIZE_BYTES = (MAX_CACHE_MEMORY_SIZE_BYTES / 4) * 3 // 75% size
+
+    private var cache: LruCache<String, ByteArray> = object :
+        LruCache<String, ByteArray>(MAX_CACHE_MEMORY_SIZE_BYTES) {
+        override fun sizeOf(key: String?, value: ByteArray): Int {
+            return value.size
+        }
+    }
+
+    override fun onTrimMemory(level: Int) {
+        when (level) {
+            ComponentCallbacks2.TRIM_MEMORY_BACKGROUND -> {
+                cache.evictAll()
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_COMPLETE -> {
+                cache.evictAll()
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_MODERATE -> {
+                cache.trimToSize(ON_MODERATE_MEMORY_SIZE_BYTES)
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL -> {
+                cache.evictAll()
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW -> {
+                cache.trimToSize(ON_LOW_MEMORY_SIZE_BYTES)
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE -> {
+                cache.trimToSize(ON_MODERATE_MEMORY_SIZE_BYTES)
+            }
+
+            ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN -> {
+                cache.evictAll()
+            }
+
+            else -> {
+                cache.evictAll()
+            }
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {}
+
+    override fun onLowMemory() {
+        cache.evictAll()
+    }
+
+    @VisibleForTesting
+    internal fun setBackingCache(cache: LruCache<String, ByteArray>) {
+        this.cache = cache
+    }
+
+    @Synchronized
+    override fun put(element: Drawable, value: String) {
+        val key = generateKey(element)
+        val byteArray = value.toByteArray(Charsets.UTF_8)
+        cache.put(key, byteArray)
+    }
+
+    @Synchronized
+    override fun get(element: Drawable): String? =
+        cache.get(generateKey(element))?.let {
+            String(it)
+        }
+
+    @Synchronized
+    override fun size(): Int = cache.size()
+
+    @Synchronized
+    override fun clear() {
+        cache.evictAll()
+    }
+
+    private fun generateKey(drawable: Drawable): String =
+        generatePrefix(drawable) + System.identityHashCode(drawable)
+
+    private fun generatePrefix(drawable: Drawable): String {
+        return when (drawable) {
+            is DrawableContainer -> getPrefixForDrawableContainer(drawable)
+            is LayerDrawable -> getPrefixForLayerDrawable(drawable)
+            else -> ""
+        }
+    }
+
+    private fun getPrefixForDrawableContainer(drawable: DrawableContainer): String {
+        if (drawable !is AnimationDrawable) {
```

Comment:
Yes, its just a fix to avoid having `AnimationDrawable` take up too much space in the lrucache. We'll need to have a discussion/brainstorming if we decide we want to send animations with base64, and this will in any case cease to be an issue once we have the resources endpoint.